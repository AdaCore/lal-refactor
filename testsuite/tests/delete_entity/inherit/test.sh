cp ./main.orig ./main.adb
delete_entity -P default.gpr -S main.adb -L 4 -R 17
echo "--  (main.orig) -L 4 -R 17"
cat ./main.adb

cp ./main.orig ./main.adb
delete_entity -P default.gpr -S main.adb -L 9 -R 28
echo "--  (main.orig) -L 9 -R 28"
cat ./main.adb
