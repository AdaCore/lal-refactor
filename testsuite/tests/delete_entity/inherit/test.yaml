title: delete_inherited

description: delete_enity tool test (on inherited decleration)

driver: shell_script
