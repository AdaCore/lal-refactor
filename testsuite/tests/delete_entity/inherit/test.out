--  (main.orig) -L 4 -R 17
procedure Main is
   package Root_Pkg is
      type Root is tagged null record;

   end Root_Pkg;

   package Child_Pkg is
      type Child is new Root_Pkg.Root with null record;

   end Child_Pkg;

   package body Child_Pkg is

   end Child_Pkg;

   package Grand_Child_Pkg is
      type Child is new Child_Pkg.Child with null record;

   end Grand_Child_Pkg;

   package body Grand_Child_Pkg is

   end Grand_Child_Pkg;

   Object : Grand_Child_Pkg.Child;
begin
   null;
end Main;
--  (main.orig) -L 9 -R 28
procedure Main is
   package Root_Pkg is
      type Root is tagged null record;
      procedure To_Be_Deleted (Self : Root) is null;
   end Root_Pkg;

   package Child_Pkg is
      type Child is new Root_Pkg.Root with null record;

   end Child_Pkg;

   package body Child_Pkg is

   end Child_Pkg;

   package Grand_Child_Pkg is
      type Child is new Child_Pkg.Child with null record;
      overriding procedure To_Be_Deleted (Self : Child);
   end Grand_Child_Pkg;

   package body Grand_Child_Pkg is
      overriding procedure To_Be_Deleted (Self : Child) is
      begin
         return;
      end To_Be_Deleted;
   end Grand_Child_Pkg;

   Object : Grand_Child_Pkg.Child;
begin
   Object.To_Be_Deleted;
end Main;
